packages/core/workflows-sdk/src/helper/workflow-export.ts

```diff
@@ -448,3 +465,38 @@ export const exportWorkflow = <TData = unknown, TResult = unknown>(
   MedusaWorkflow.registerWorkflow(workflowId, exportedWorkflow)
   return exportedWorkflow as MainExportedWorkflow<TData, TResult>
 }
+
+function attachOnFinishReleaseEvents(
```

Comment:
question: Just to be sure I understand, this means that releasing grouped events happens separately from the workflow execution, i.e. in case emitting the grouped events fails, the compensation flow is **not** run?

Reply:
Yes thats right, do you think i should instead create a step instead of listening to finish ? Basically applying the release step at the end of the workflow ?

Cc @carlos-r-l-rodrigues

Reply:
> do you think i should instead create a step instead of listening to finish ?

This is what I had in mind. My assumption is that most times, the event emission done in the workflow is critical to the operation performed, so in case the emission fails, I would expect it to be rolled back and we can try again.

Reply:
Either we have it as a separeted step, or we explictly cancel the flow with `transaction.cancel()` if the emit fails.
It depends if we find a problem to display this new magic step in the UI for example.

Reply:
what do you think would be the most appropriate guys?

Reply:
If we can identify the final injected release events step, so we can hide it in the UI (if we want to), then I am leaning towards that approach. The reasoning here is that then we rely on an existing "native" workflow mechanism, instead of having an additional, separate path that needs testing and maintenance. 

Wdyt Carlos?

Reply:
I prefer to not have the magic step, unless it is explicity added. Otherwise we also need some other complex changes to make sure this will be the last step even if the workflow is edited.
I would just handle the emitGroupedEvents and add cancel the transaction if that fails, and add it to the errors returned.

Reply:
Ah OK, so we can still cancel the workflow in the `onFinish`? I thought it was after the execution was done.

Reply:
I ll go with the cancel guys then. First I need to finish my other future pr that I am on at the moment for another part of the group id management 😂